### PR TITLE
New section Firmware Releases

### PR DESCRIPTION
**Pull request by technobly**
Adds these sections to Firmware Reference in Docs:
- Firmware Releases
  - Firmware Release Process
  - Github Release Notes
  - Programming and Debugging Notes

Github Release pages will now link to the Programming and Debugging Notes section with a dynamic URL such as https://docs.particle.io/reference/firmware/photon/?fw_ver=0.5.3&cli_ver=1.17.0&electron_parts=2#programming-and-debugging-notes instead of statically created for each firmware release.

### CODE REVIEW

File: templates/partials/scripts.hbs
Comment (jme783) [inline templates/partials/scripts.hbs]: Would it be possible to load this script on only the pages you need instead of on every one? Sometimes weird JS things happen when some jQuery expects content that isn't there

**Review by jme783 — APPROVED** [review]
Overall, looks like this will do the trick! Using jQuery in this way makes me a tad bit nervous from a maintainability and rigidity standpoint, but this seems to be a short-term solution to unblock the firmware team.

In the long term, I am very excited about coming up with a flexible and straightforward way to add versioning to our documentation.

**Comment by monkbroc** [comment]
Let's not hold up the release but I did mention to @stefan-moraru that it should be possible to use a Handlebar block helper like this instead of using markers that will be removed by jQuery. We just haven't gotten around to pairing on this...

```
{{#section "release-notes-0.5.3}}
# Notes
More markdown
{{/section}}
```

File: templates/partials/scripts.hbs
Comment (technobly) [reply templates/partials/scripts.hbs]: Stefan and I got that worked out @jme783 so now the script will only load on reference pages and only run on the firmware.md page.

**Review by stefan-moraru — APPROVED** [review]
Looks good

**Comment by technobly** [comment]
Most of the latest firmware releases and prereleases now link back to this new Docs feature: https://github.com/spark/firmware/releases

**Pull request merged**
